### PR TITLE
[Bugfixes] Multiple alignment/button bugfixes

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested all changes
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Some buttons in Artemis are not aligned correctly.
Other not necessary buttons are removed.
Also the "Correction Round 1" Column is changed to "Assessment Action" in all assessment dashboards.

### Description
1. Center align of the Test Run button:
  -->  

2. Replace "Correction Round 1" with "Assessment actions" When there is only one round.
 -->  

3. Change Buttons in the exercise overview: remove "View" Button for all types as you can simply click on the id/title of the exercise. Also modified programming exercise buttons to look cleaner.
image
image

4. Changed the "Test run ribbon".
 -->  

### Steps for Testing
1. Go to the exam overview of a course -> Test Runs -> see if the "Assess your testrun" button is vertically centered
2. Check in the [exercise-type]-exercise-assessment-dashboard that the last column(s) are correct according to the number of correctionrounds.
3. Check the buttons in the course -> exercise overview.
4.  Go to an exam and check the test run ribbon, and the test run summary ribbons.

### CODE REVIEW

**Review by nicolasruscher — COMMENTED** [review]
works locally 👍  will do code review later :D

**Review by airelawaleria — APPROVED** [review]

**Review by simonzilker — APPROVED** [review]
Looks good

**Review by marvinchimeka — APPROVED** [review]
No issues found

**Review by JohannesWeiss — DISMISSED** [review]

**Review by derLalla — CHANGES_REQUESTED** [review]
those are really nice changes! I have only one point which you could maybe address.

The alignment of the buttons in the Exam Management is off (Click on "Student Exams"):

image

File: src/main/webapp/app/exercises/programming/assess/programming-assessment-dashboard/programming-assessment-dashboard.component.html [hunk 1]
```diff
@@ -30,12 +30,17 @@ <h4 [translateValues]="{ exerciseTitle: exercise.title }" jhiTranslate="artemisA
...
-                    <th *ngIf="numberOfCorrectionrounds == 2 && newManualResultAllowed">
-                        <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.assessmentSecond">Assessment</a>
-                    </th>
+                    <ng-container *ngIf="newManualResultAllowed">
+                        <th *ngIf="exercise.course || (exercise.exerciseGroup && numberOfCorrectionrounds == 1)">
+                            <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.action">Assessment</a>
+                        </th>
+                        <th *ngIf="numberOfCorrectionrounds == 2">
+                            <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.assessmentFirst">Assessment</a>
+                        </th>
```
Comment (derLalla) [inline src/main/webapp/app/exercises/programming/assess/programming-assessment-dashboard/programming-assessment-dashboard.component.html:39]: is this a mistake and you meant to have `== 1`?
```suggestion
                        <th *ngIf="numberOfCorrectionrounds == 1">
                            <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.assessmentFirst">Assessment</a>
                        </th>
```

File: src/main/webapp/app/exercises/text/assess/text-assessment-dashboard/text-assessment-dashboard.component.html [hunk 2]
```diff
@@ -39,7 +35,10 @@ <h4 [translateValues]="{ exerciseTitle: exercise.title }" jhiTranslate="artemisA
                         <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.assessor">Reviewer</a>
                         <fa-icon [icon]="'sort'"></fa-icon>
                     </th>
-                    <th>
+                    <th *ngIf="exercise.course || (exercise.exerciseGroup && numberOfCorrectionrounds == 1)">
+                        <a class="th-link" jhiTranslate="artemisApp.assessment.dashboard.columns.action">Assessment</a>
+                    </th>
+                    <th *ngIf="numberOfCorrectionrounds == 2">
```
Comment (derLalla) [inline src/main/webapp/app/exercises/text/assess/text-assessment-dashboard/text-assessment-dashboard.component.html:41]: also here? `== 1 `

```suggestion
                    <th *ngIf="numberOfCorrectionrounds == 1">
```

**Review by derLalla — CHANGES_REQUESTED** [review]
I have looked through the code and left some comments.

**Review by tiffca — DISMISSED** [review]
Tested successfully

File: src/main/webapp/app/exercises/programming/assess/programming-assessment-dashboard/programming-assessment-dashboard.component.html (hunk 1 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/programming/assess/programming-assessment-dashboard/programming-assessment-dashboard.component.html:39]: No, here i need to check if there numberOfCorrectionRounds is 2. (If there are 2, I want to display the column title for the first correction, and for the second correction)

File: src/main/webapp/app/exercises/text/assess/text-assessment-dashboard/text-assessment-dashboard.component.html (hunk 2 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/text/assess/text-assessment-dashboard/text-assessment-dashboard.component.html:41]: same here

**Review by nicolasruscher — DISMISSED** [review]
code looking good!

**Review by EneaGore — APPROVED** [review]
Could not find any issues on ts1

**Review by sebastiantpeter — APPROVED** [review]
Looks good on ts1.

**Comment by krusche** [comment]
> those are really nice changes! I have only one point which you could maybe address.
> 
> The alignment of the buttons in the Exam Management is off (Click on "Student Exams"):
> 
> image

we'll fix this in a follow-up

**Review by krusche — CHANGES_REQUESTED** [review]
there are some Codacy errors in the scss which should be fixed

File: src/main/webapp/app/shared/layouts/profiles/page-ribbon.scss [hunk 3]
```diff
@@ -1,31 +1,52 @@
...
-    top: 40px;
     white-space: nowrap;
-    width: 15em;
-    z-index: 9999;
     pointer-events: none;
+    position: absolute;
     opacity: 0.75;
-    a {
-        color: #fff;
+    span {
```
Comment (SimonEntholzer) [inline src/main/webapp/app/shared/layouts/profiles/page-ribbon.scss:23]:
```suggestion
    
    span {
```

File: src/main/webapp/app/shared/layouts/profiles/page-ribbon.scss [hunk 4]
```diff
@@ -1,31 +1,53 @@
...
-    top: 40px;
     white-space: nowrap;
-    width: 15em;
-    z-index: 9999;
     pointer-events: none;
+    position: absolute;
     opacity: 0.75;
-    a {
-        color: #fff;
+
```
Comment (SimonEntholzer) [inline src/main/webapp/app/shared/layouts/profiles/page-ribbon.scss:23]:
```suggestion

```

**Review by airelawaleria — DISMISSED** [review]
re-approve

**Review by JohannesWeiss — DISMISSED** [review]
Re-Approve

**Review by LudwigWeide — APPROVED** [review]

**Review by simonzilker — DISMISSED** [review]
Re-approve

**Review by marvinchimeka — APPROVED** [review]

**Review by FrankeLukas — APPROVED** [review]
Generally looks good to me, well done! One thing I noticed that looks a bit weird imo:
grafik
Why are we displaying "Test Run" for every exercise in the exam summary?

**Comment by SimonEntholzer** [comment]
> Generally looks good to me, well done! One thing I noticed that looks a bit weird imo:
> grafik
> Why are we displaying "Test Run" for every exercise in the exam summary?

Before the change there was a second ribbon overlaying the first "Test Run" ribbon in the same place. I thought in these places it would make more sense.

**Review by FrankeLukas — DISMISSED** [review]
Re-approve.

**Review by airelawaleria — APPROVED** [review]

**Review by marvinchimeka — APPROVED** [review]

**Review by LudwigWeide — APPROVED** [review]

**Review by simonzilker — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]

**Review by nicolasruscher — APPROVED** [review]
Re-approve

**Review by FrankeLukas — APPROVED** [review]
Re-approve.

**Review by krusche — APPROVED** [review]

**Pull request merged**